in_toto/in_toto_run.py

```diff
@@ -142,23 +155,44 @@ def main():
   # Override defaults in settings.py with environment variables and RCfiles
   in_toto.user_settings.set_settings()
 
+  # Regular signing and GPG signing are mutually exclusive
+  if (args.key == None) == (args.gpg == None):
+    parser.print_usage()
+    parser.exit("Specify either `--key <key path>` or `--gpg [<keyid>]`")
+
+  # If `--gpg` was set without argument it has the value `True` and
+  # we will try to sign with the default key
+  gpg_use_default = (args.gpg == True)
+
+  # Otherwise we interpret it as actual keyid
+  gpg_keyid = None
+  if args.gpg != True:
+    gpg_keyid = args.gpg
+
+
   # We load the key here because it might prompt the user for a password in
   # case the key is encrypted. Something that should not happen in the library.
-  try:
-    key = util.prompt_import_rsa_key_from_file(args.key)
-  except Exception as e:
-    log.error("in load key - {}".format(args.key))
-    sys.exit(1)
+  key = None
+  if args.key:
+    try:
+      key = util.prompt_import_rsa_key_from_file(args.key)
+
+    except Exception as e:
+      log.error("in load key - {}".format(args.key))
+      sys.exit(1)
 
+  # If no_command is specified run in_toto_run without executing a command
   if args.no_command:
     in_toto_run(args.step_name, args.materials, args.products, [],
-      key, args.record_streams)
+      args.record_streams, key, gpg_keyid, gpg_use_default, args.gpg_home)
+
   else:
     if not args.link_cmd:
       parser.print_usage()
-      parser.exit("For no command use --no-command option")
-    in_toto_run(args.step_name, args.materials, args.products,
-      args.link_cmd, key, args.record_streams)
+      parser.exit("For no command use `--no-command` option")
```

Comment:
can we rephrase as follows:
```
ERROR: No command specified. Please specify (or use the --no-command option)
```
It feels a little bit clearer to me.

Reply:
I'm wondering if refactoring this to
```python
if args.no_command:
  args.link_cmd = []
elif not args.link_cmd:
  parser.print_usage()
  parser.exit("...")

in_toto_run(...)
```
Would make this piece of code more readable.

Reply:
can do both